### DISCUSSION TITLE
Changing RTSP port bricks camera.

### DISCUSSION THREAD

**Discussion by sierraworks** [gtxaspec/wz_mini_hacks#641] [Q&A]
Cameras:    1 V3 cam and 1 Pan V2.  

Running the latest mini_hacks but I didn't log the firmware versions.    

Steps to reproduce:

1:  get wz_mini_hacks working on the camera, verify with web interface, ssh, etc.
2:  change the RTSP port from 8554 to 554 and reboot.
3:  On the V3 camera the red light comes on but the camera is dead, no ping, no arp, nothing.
     On the PanV2 the light is purple and no ping, etc.
4:  Remove SD card, power cycle - still nothing 
5:  Hold down setup (with and without power or when adding power and with or with SD card) - nothing
6:  Put latest firmware on SD card and try to manually flash --  nothing

It's an interesting puzzle.   The cameras were working fine before I changed the port.

Anyone else want to try to brick their cameras?

**Comment by gtxaspec** [comment]
How are you changing port, via ssh or web?
Also what version of wz mini are you running?

**Reply by sierraworks** [reply]
I'm on master as of yesterday.    I'll try another erase SD and start over but this is what I'm currently seeing.

I set the port on the config screen and then set it back by pulling the SD and manually changing the config file.   I removed the logs to see if anything is running and I do get this far:

------   From /logs/wz_init.log   (actually mounted on my mac as /Volumes/WZ3/wz_mini/log

#####S00config#####
No Windows line endings detected. Configuration file is valid.
#####S01bind#####
...
#####S04model#####
cat: can't open '/opt/wz_mini/tmp/configs/.product_config': No such file or directory
 detected
#####S09firstboot#####
First boot already completed
#####S11configbackup#####
Factory config backup directory exists
T31 platform backup
Backup /configs
#####S15fwupdate#####
Current stock firmware version:
[VER]
appver=4.49.0.97
#####S30patchicamera#####
+ mount -t squashfs /dev/mtdblock3 /system
mount: mounting /dev/mtdblock3 on /system failed: Resource busy
+ mount -t jffs2 /dev/mtdblock6 /configs
#####S01swap#####
Swap archive missing, not extracting
Swap file found, enable
ifconfig: wlan0: error fetching interface information: Device not found
K01network: Network HW not ready yet, try again in 5 seconds
S09coredump: wpa_supplicant not initialized yet, try again in 5 seconds
#####S10firmware#####
Firmwware updates enabled, monitor script running
#####S14nightdrop#####
#####S18fps#####
System FPS greater than default, setting.
cat: can't open '/configs/.user_config': No such file or directory
sed: /configs/.user_config: No such file or directory
#####S19video#####
+ /opt/wz_mini/tmp/.storage/app_init.sh
------

**Reply by sierraworks** [reply]
Doing a fresh retry:

1:  empty the SD Card,  verify with  ls -la  /Volumes/<sd card name>  
2:  copy the SD dir on mini hacks master branch
3:  try to startup.

Now the log is pretty short and is below.   What's interesting is that it appears that the wireless gets hosed (somehow).    I tried a factory reset --   (no power hold down 'setup'),  holding setup when applying power,  apply power and then hold setup  ( all without the SD card).

Nothing seems to work.

--------  log

+ export 'PATH=/bin:/sbin:/usr/bin:/usr/sbin:/opt/wz_mini/bin'
+ export 'PATH=/system/bin:/bin:/sbin:/usr/bin:/usr/sbin:/opt/wz_mini/bin'
+ export 'LD_LIBRARY_PATH=/system/lib:/opt/wz_mini/lib'
+ export 'LD_LIBRARY_PATH=/thirdlib:/system/lib:/opt/wz_mini/lib'
+ ifconfig lo up
+ mount -t squashfs /dev/mtdblock3 /system
mount: mounting /dev/mtdblock3 on /system failed: Resource busy
+ mount -t jffs2 /dev/mtdblock6 /configs
+ /opt/wz_mini/etc/rc.d/K03rcd
+ /opt/wz_mini/etc/rc.d/K01network
#####S01swap#####
insmod: can't insert '/opt/wz_mini/lib/modules/3.10.14__isvp_swan_1.0__/extra/audio.ko': invalid parameter
Segmentation fault
rmmod: remove 'audio': No such file or directory
Swap archive present, extracting
ifconfig: wlan0: error fetching interface information: Device not found
K01network: Network HW not ready yet, try again in 5 seconds
ifconfig: wlan0: error fetching interface information: Device not found
K01network: Network HW not ready yet, try again in 5 seconds
ifconfig: wlan0: error fetching interface information: Device not found
K01network: Network HW not ready yet, try again in 5 seconds
Setting up swapspace version 1, size = 134213632 bytes
UUID=52f34759-5f92-47b9-b365-0a97941cc8d4
Swap file found, enable
#####S03ethernet#####
USB Ethernet disabled
#####S04usbdirect#####
USB Direct disabled
USB RNDIS disabled
No alternate iface enabled
S09coredump: wpa_supplicant not initialized yet, try again in 5 seconds
#####S10firmware#####
Firmwware updates enabled, monitor script running
#####S14nightdrop#####
#####S17motor#####
Motor enabled
#####S18fps#####
sh: 20: unknown operand
#####S19video#####
+ /opt/wz_mini/tmp/.storage/app_init.sh
ifconfig: wlan0: error fetching interface information: Device not found
K01network: Network HW not ready yet, try again in 5 seconds

**Reply by gtxaspec** [reply]
@virmaior looks like the config got clobbered... `opt/wz_mini/tmp/configs/.product_config': No such file or directory`

Any idea why?

**Reply by gtxaspec** [reply]
@sierraworks looks like the modules are missing from the card, can you verify in the SD root in lib/modules/3.10.14__swan/extra?

**Reply by sierraworks** [reply]
They appear to be the same as a working camera:
ls /Volumes/WYZE03/wz_mini/lib/modules/3.10.14__isvp_swan_1.0__/extra/
8189fs.ko			usbcamera.ko   atbm603x_wifi_sdio.ko		v4l2loopback.ko
atbm_txpwer_dcxo_cfg.txt	videobuf2-vmalloc.ko   audio.ko

However on a fresh install -  i.e.   copy files from master/SD/ to a card and boot also fails so I'm thinking something screwed up the hardware. 

I'm going to move on and just get a new Pan2 and not change the RTSP port.     My V3 doesn't do anything on boot now, including touching the SD card so I think that's a lost cause.

Thanks for the feedback though.

**Reply by gtxaspec** [reply]
It can be fixed, do you have the configs backup folder on your SD card?

**Reply by gtxaspec** [reply]
@virmaior didn't we have an easy way to restore the configs? i can't find it now, though I'm on mobile.

**Reply by virmaior** [reply]
@gtxaspec  https://github.com/gtxaspec/wz_mini_hacks/issues/329

**Reply by sierraworks** [reply]
Thanks for the link,  sadly I've reformatted and even replaced the SD card (multiple times).   I have a Pan2 coming today and I'll install on it,  then I'll take the card from there and try it in the non-responsive camera.    If that doesn't fix it then I'll just trash it and move on.   Again thanks to everyone for the assistance.

**Comment by virmaior** [comment]
@gtxaspec  I'll check through the code, but my running hypothesis is that the issue is not from the port change itself but from the `reboot` command issued to the camera as part of the process. (i.e., I think I must be using the wrong command to issue the reboot (`reboot`) or there's something janky that can happen on wyze cameras when rebooting, because I periodically lose `/configs` on cameras without changing settings when rebooting).  

I'll check through my ash code a bit later and see if there's anything that could cause it.

(case in point, one of my cameras clobbered its /configs last night at approximately 9:32pm without any apparent reason; best I can guess that would have been either from viewing the screenshot or from the irled turning on).

**Reply by sierraworks** [reply]
Thanks,  I have one last working camera,  (v3)  I'll try to scp -r /configs and see if that fixes it.

**Comment by vic4news** [comment]
This happened to me on two different v3 cameras when changing Rtsp settings, but not the port. 
The cams weren’t bricked but they did lose the WiFi settings and would have a flashing red light when powered on. To fix them, remove the sd card, power on the cam and the red light will be blinking. Use the wyze app to add the camera by pressing the + sign to add a device, then press the setup button on the cam and you will be prompted to setup the camera.

I could then put the sd card back in the cam and mini hacks would load normally.

**Comment by archandanime** [comment]
I did the following on my Wyze cam v3:
- Download latest wz_mini_hacks, copt SD_ROOT to SD card, replace SSH authorized_keys
- Boot up the camera
- SSH to the camera and enable RTSP:
```
RTSP_HI_RES_ENABLED="true"
```
- Reboot and confirm RTSP is running:
```
> nmap 192.168.10.20 -p554,8554

PORT     STATE  SERVICE
554/tcp  closed rtsp
8554/tcp open   rtsp-alt
```
- SSH to the camera and change RTSP port
```
RTSP_PORT="554"
```
- Reboot and check opened ports on the camera
```
> nmap 192.168.10.20 -p554,8554

PORT     STATE  SERVICE
554/tcp  open   rtsp
8554/tcp closed rtsp-alt
```
- Check Wyze app, the camera live feed is normal.

Hmmm I have no idea.

**Reply by virmaior** [reply]
1. I think the people who are reporting this are using the web interface to change the setting. (So far I don't really know how/why changing via the web interface would lead to this).
2. I've actually seen /configs get clobbered on my cameras without changing anything (usually I figure out its clobbered because TZ disappears and the recording gets shifted off by the time difference)
3. My current hypothesis is that this has something to do with the camera not getting enough power (my cameras are all powered by a usb hub and run ethernet over usb -- this issue only occurs with a couple of the cameras) and/or an issue occurring during reboot.

How do you power your cameras? what is the exact command you're using to reboot (I use `reboot`).

**Reply by archandanime** [reply]
For power supply, I use a 5V-2.1A wall charger, it has only one power output port.
I power on the camera by plugging the camera cord to the wall changer, I am not aware of another way.
For reboot command, I use `reboot` too.

**Reply by archandanime** [reply]
Someone [posted](https://forums.wyze.com/t/power-requirement-for-v3-cam/181468/6) on Wyze forum that the v3 camera draws 0.387A with spotlight on.
A standard USB port supplies 0.5A, you sad a USB hub, not a dock, right? if multiple cameras share a USB hub, the current may be insufficient.

**Reply by virmaior** [reply]
I'm using a powered-usb hub but I'll need to check the numbers on power output. It seems to function fine *most* of the time. (I guess it's mostly that I'm struggling to explain how anything in the [b]ash code for the web server would encourage `/config` getting wiped -- since we killed any modification of `/configs` via web as basically to error prone).

**Reply by gtxaspec** [reply]
do we read from /configs? maybe something there?

**Reply by virmaior** [reply]
@gtxaspec 

(1) I've got this blocked in `shared.cgi`

```
if [ -f /opt/wz_mini/tmp/.T31 ]; then
  camtype=T31
  camfirmware=$(tail -n1 /configs/app.ver | cut -f2 -d=  )
  base_cam_config="/configs/.user_config"
elif [ -f /opt/wz_mini/tmp/.T20 ]; then
  camtype=T20
  camfirmware=$(tail -n1 /system/bin/app.ver | cut -f2 -d= )
  base_cam_config="/configs/.parameters"
fi
cam_config=$base_cam_config
```
but that's only used in `cam.cgi` (where the form is completely disabled).

(2) `status.cgi`

has `TZ=$(cat /configs/TZ)`


I don't see any other code that uses /configs in the web area

**Reply by sierraworks** [reply]
When my new camera arrives I'll give the ssh RTSP change a try.   I also just use $ reboot as well for restarting.
